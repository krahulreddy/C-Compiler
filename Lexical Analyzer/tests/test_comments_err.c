#include <stdio.h>

int sum(int a, int b)
{
    int sum = a + b;
    /*
    This is an example of a function
        /*
        The function here is used to add two numbers
    */

    return(sum);
}

int main()
{
    int a = 10, b = 6; //Declaring Vairables
    /*
    INT - Keyword
    a & b - Identifiers
    */

   //Getting the sum of the numbers
    int ans = sum(a,b);

    //Printing the answer
    printf("Sum : %d", ans);

    return(0);
}