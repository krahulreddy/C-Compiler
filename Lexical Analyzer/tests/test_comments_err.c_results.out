Line			Lexeme			Token Class
1		#include <stdio.h> 		Pre Processor directive

2
3			int 			KEYWORD
			sum 			IDENTIFIER
			( 			OPENING BRACKETS
			int 			KEYWORD
			a 			IDENTIFIER
			, 			COMMA DELIMITER
			int 			KEYWORD
			b 			IDENTIFIER
			) 			CLOSING BRACKETS

4			{ 			OPENING BRACES

5			int 			KEYWORD
			sum 			IDENTIFIER
			= 			OPERATOR
			a 			IDENTIFIER
			+ 			OPERATOR
			b 			IDENTIFIER
			; 			SEMICOLON DELIMITER

6			/*
    This is an example of a function
        /*
        The function here is used to add two numbers
    */ 			MULTI LINE COMMENT

7XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
ERROR: at line no. 7
	+
	^^
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX


SYMBOL TABLE

	a			IDENTIFIER
	b			IDENTIFIER
	int			KEYWORD
	sum			IDENTIFIER


CONSTANT TABLE

